### ISSUE TITLE
Consumer can't find the metadata of provider when using remote metadata service

### ISSUE THREAD

**Issue by Patrick0308** [apache/dubbo#7027]
- [ ] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.
- [ ] I have checked the [FAQ](https://github.com/apache/dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 2.7.7
* Operating System version: Mac OS
* Java version: 1.8

### Steps to reproduce this issue

This is simple demo to reproduce issue:
https://github.com/Patrick0308/dubbo-samples/tree/remote-metadata-service/dubbo-samples-cloud-native/dubbo-demo-servicediscovery-xml

### Expected Result

Consumer can invoke successfully.

### Actual Result

```
Exception in thread "main" org.apache.dubbo.rpc.RpcException: Failed to invoke the method sayHello in the service org.apache.dubbo.demo.DemoService. No provider available for the service org.apache.dubbo.demo.DemoService from registry 127.0.0.1:2181 on the consumer 192.168.0.101 using the dubbo version 2.7.7. Please check if the providers have been started and registered.
	at org.apache.dubbo.rpc.cluster.support.AbstractClusterInvoker.checkInvokers(AbstractClusterInvoker.java:282)
	at org.apache.dubbo.rpc.cluster.support.FailoverClusterInvoker.doInvoke(FailoverClusterInvoker.java:59)
	at org.apache.dubbo.rpc.cluster.support.AbstractClusterInvoker.invoke(AbstractClusterInvoker.java:259)
	at org.apache.dubbo.rpc.cluster.interceptor.ClusterInterceptor.intercept(ClusterInterceptor.java:47)
	at org.apache.dubbo.rpc.cluster.support.wrapper.AbstractCluster$InterceptorInvokerNode.invoke(AbstractCluster.java:92)
	at org.apache.dubbo.rpc.cluster.support.wrapper.MockClusterInvoker.invoke(MockClusterInvoker.java:82)
	at org.apache.dubbo.rpc.proxy.InvokerInvocationHandler.invoke(InvokerInvocationHandler.java:74)
	at org.apache.dubbo.common.bytecode.proxy0.sayHello(proxy0.java)
	at org.apache.dubbo.demo.consumer.Application.main(Application.java:32)
```

**Comment by cityiron** [comment]
When I use Zookeeper or Consul is not run successful too.

```bash
org.apache.dubbo.rpc.RpcException: Failed to invoke the method GetUsers in the service org.apache.dubbo.UserProvider. No provider available for the service org.apache.dubbo.UserProvider from registry 127.0.0.1:2181 on the consumer 127.0.0.1 using the dubbo version 2.7.7. Please check if the providers have been started and registered.
	at org.apache.dubbo.rpc.cluster.support.AbstractClusterInvoker.checkInvokers(AbstractClusterInvoker.java:282)
	at org.apache.dubbo.rpc.cluster.support.FailoverClusterInvoker.doInvoke(FailoverClusterInvoker.java:59)
	at org.apache.dubbo.rpc.cluster.support.AbstractClusterInvoker.invoke(AbstractClusterInvoker.java:259)
	at org.apache.dubbo.rpc.cluster.interceptor.ClusterInterceptor.intercept(ClusterInterceptor.java:47)
	at org.apache.dubbo.rpc.cluster.support.wrapper.AbstractCluster$InterceptorInvokerNode.invoke(AbstractCluster.java:92)
	at org.apache.dubbo.rpc.cluster.support.wrapper.MockClusterInvoker.invoke(MockClusterInvoker.java:82)
	at org.apache.dubbo.rpc.proxy.InvokerInvocationHandler.invoke(InvokerInvocationHandler.java:74)
	at org.apache.dubbo.common.bytecode.proxy0.GetUsers(proxy0.java)
	at org.apache.dubbo.Consumer.testGetUsers(Consumer.java:101)
	at org.apache.dubbo.Consumer.start(Consumer.java:47)
	at org.apache.dubbo.Consumer.main(Consumer.java:40)
```

**Comment by chickenlj** [comment]
Fixed in 2.7.8
